File: Java/Migration/unit-tests-csharp/unit-tests-csharp/P03Any/Ex02GenericAny.cs
using NUnit.Framework;
using TddEbook.TddToolkit;

namespace unit_tests_csharp.P03Any
{
  public class Ex02GenericAny
  {
    [Test]
    public void ShouldCreateGenericInstances()
    {
      //WHEN
      MyClass<int> anonymous = Any.Instance<MyClass<int>>();

      //THEN
      Assert.AreNotEqual(2, anonymous.Instance); //TODO change to AreEqual()
    }
  }
}

File: Java/Migration/unit-tests-java/src/test/java/p03_any/Ex02GenericAny.java
package p03_any;

import autofixture.publicinterface.Any;
import autofixture.publicinterface.InstanceOf;
import org.testng.annotations.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class Ex02GenericAny {
  @Test
  public void shouldCreateGenericInstances() {
    //WHEN
    MyClass<Integer> anonymous = Any.anonymous(new InstanceOf<MyClass<Integer>>() {});

    //THEN
    assertThat(anonymous.instance).isNotEqualTo(2); //TODO change to isEqualTo()
  }
}

